### PR TITLE
Updating evaluation script

### LINKED ISSUE #109: number of classes error in accuracy plots
When `affinity.csv` is not present, the accuracy_plot throws the following error:

`ValueError: The number of FixedLocator locations (13), usually from a call to set_ticks, does not match the number of labels (9).`

### PR DESCRIPTION
**Pull request by crangelsmith**
Fixes #109 #114 #122 #126

- [x] Tracks labels of evaluation data (or labels it as 'test' if no info is provided)
- [x] Adds prediction and accuracy plots on evaluation.
- [x] Plots embeddings in the same figure to the training set for comparisons.
- [x] Adds option to not overwrite plots (you can pass extra text for the file name in function).
- [x] Fix bug on affinity matrix selection for subset of classes (#122 )
- [x] Fix bugs on dynamic plots (#114)
- [x] Add randomisation seed in test-train split and embeddings for reproducibility.
- [x] Adding docstrings to functions

### CODE REVIEW

**Review by marjanfamili — APPROVED** [review]
runs fine for me. Thanks

File: avae/evaluate.py [hunk 1]
```diff
@@ -31,12 +53,13 @@ def evaluate(datapath, lim, splt, batch_s, collect_meta, use_gpu):
     vae.to(device)
 
     # ########################## EVALUATE ################################
-    if collect_meta:
```
Comment (crangelsmith) [inline avae/evaluate.py:34]: Will be bringing this back due to issue raised by @marjanfamili

File: avae/evaluate.py [hunk 2]
```diff
@@ -65,15 +94,50 @@ def evaluate(datapath, lim, splt, batch_s, collect_meta, use_gpu):
     if config.VIS_REC:
         vis.recon_plot(x, x_hat, name="evl")
 
+    # get training latent space
+    latents_training = meta_df[meta_df["mode"] == "trn"][
```
Comment (crangelsmith) [inline avae/evaluate.py:98]: Should be reading latent space from `.pt` file and not `.pkl` file, will fix this before merging.

File: avae/evaluate.py (hunk 1 above)
Comment (crangelsmith) [reply avae/evaluate.py:34]: Fixed in [1445299](https://github.com/alan-turing-institute/affinity-vae/pull/120/commits/1445299eeb021ff0597978e5e72fad30f5d2bac5)

**Review by jolaem — APPROVED** [review]

**Pull request merged**